diff --git a/arch/arm/mach-ux500/board-pydtd.c b/arch/arm/mach-ux500/board-pydtd.c
index c2aa1e7..92448ed 100644
--- a/arch/arm/mach-ux500/board-pydtd.c
+++ b/arch/arm/mach-ux500/board-pydtd.c
@@ -41,6 +41,8 @@
 #include <mach/board_htc.h>
 /* HTC_PROC_EMMC_END */
 
+#define AZHE
+
 #ifdef CONFIG_USB_ANDROID
 #include <linux/usb/android_composite.h>
 #include <mach/htc_usb.h>
@@ -746,6 +748,28 @@ static struct mhl_platform_data mhl_sii9232_device_data = {
 };
 #endif /* CONFIG_STE_HDMI_MHL */
 
+#ifdef AZHE
+static struct bma250_platform_data bma250_data = {
+	.intr = 171,
+	.chip_layout = 1,
+	.calibration_mode = 0,
+	.gs_kvalue = 0,
+};
+
+#define AKM8975_COMPASS_LAYOUTS {\
+			{ { 0,  1, 0}, {-1,  0,  0}, {0, 0,  1} }, \
+	                { { 0, -1, 0}, { 1,  0,  0}, {0, 0, -1} }, \
+	                { {-1,  0, 0}, { 0, -1,  0}, {0, 0,  1} }, \
+	                { { 1,  0, 0}, { 0,  0,  1}, {0, 1,  0} }  \
+}
+
+static struct akm8975_platform_data akm8975_data = {
+	.layouts = AKM8975_COMPASS_LAYOUTS,
+	.irq_trigger = 0,
+};
+
+#endif
+
 static struct i2c_board_info __initdata pydtd_i2c0_devices[] = {
 	{
 		I2C_BOARD_INFO(MICROP_I2C_NAME, 0xCC >> 1),
@@ -762,6 +786,19 @@ static struct i2c_board_info __initdata pydtd_i2c0_devices[] = {
 		.irq = GPIO_TO_IRQ(199),
 		.platform_data = &mpu3050_data,
 	},
+#ifdef AZHE
+	{
+		I2C_BOARD_INFO(BMA250_I2C_NAME,0x30>>1),
+		.irq = GPIO_TO_IRQ(171),
+		.platform_data = &bma250_data,
+		
+	},
+	{
+		I2C_BOARD_INFO(AKM8975_I2C_NAME,0x1A>>1),
+		.irq = GPIO_TO_IRQ(153),
+		.platform_data = &akm8975_data,
+	},
+#endif
 #ifdef CONFIG_TPS65200
 	{
 		I2C_BOARD_INFO("tps65200", 0xD4 >> 1),
diff --git a/drivers/i2c/chips/akm8975.c b/drivers/i2c/chips/akm8975.c
index 64ada25..8f791d5 100644
--- a/drivers/i2c/chips/akm8975.c
+++ b/drivers/i2c/chips/akm8975.c
@@ -12,6 +12,9 @@
  * GNU General Public License for more details.
  */
 
+#define DEBUG
+#define pr_fmt(fmt)	"[azhe][AKM8975]" fmt
+
 #include <linux/interrupt.h>
 #include <linux/i2c.h>
 #include <linux/slab.h>
@@ -26,7 +29,7 @@
 #include <linux/akm8975.h>
 #include<linux/earlysuspend.h>
 
-#define DEBUG 0
+#define DEBUG 1
 #define MAX_FAILURE_COUNT 3
 /*#define AKM_EARLY_SUSPEND 1*/
 
@@ -1032,7 +1035,7 @@ int akm8975_probe(struct i2c_client *client, const struct i2c_device_id *id)
 	atomic_set(&t_flag, 0);
 	atomic_set(&mv_flag, 0);
 
-	debug_flag = 0;
+	debug_flag = 1;
 	debug_flag_fatal_err = 0;
 	fatal_err_pr_count = 0;
 
@@ -1059,7 +1062,8 @@ int akm8975_probe(struct i2c_client *client, const struct i2c_device_id *id)
 		E("%s: request irq failed\n", __func__);
 		goto exit_irq_request_failed;
 	}
-
+	
+	pr_debug("%s:ok!\n",__func__);
 	return 0;
 
 exit_irq_request_failed:
@@ -1074,6 +1078,8 @@ exit_platform_data_null:
 	kfree(akm);
 exit_alloc_data_failed:
 exit_check_functionality_failed:
+
+	pr_debug("%s:faild!\n",__func__);
 	return err;
 
 }
diff --git a/drivers/i2c/chips/bma250.c b/drivers/i2c/chips/bma250.c
index 0078d25..a9289b4 100644
--- a/drivers/i2c/chips/bma250.c
+++ b/drivers/i2c/chips/bma250.c
@@ -11,6 +11,7 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
  */
+#define pr_fmt(fmt) "[azhe][BMA250]" fmt
 
 #include <linux/i2c.h>
 #include <linux/slab.h>
@@ -279,16 +280,16 @@ static int bma_ioctl(struct inode *inode, struct file *file, unsigned int cmd,
 	case BMA_IOCTL_READ:
 		if (rwbuf[0] < 1)
 			return -EINVAL;
-		/*ret = BMA_I2C_RxData(&rwbuf[1], rwbuf[0]);
+		ret = BMA_I2C_RxData(&rwbuf[1], rwbuf[0]);
 		if (ret < 0)
-			return ret;*/
+			return ret;
 		break;
 	case BMA_IOCTL_WRITE:
 		if (rwbuf[0] < 2)
 			return -EINVAL;
-		/*ret = BMA_I2C_TxData(&rwbuf[1], rwbuf[0]);
+		ret = BMA_I2C_TxData(&rwbuf[1], rwbuf[0]);
 		if (ret < 0)
-			return ret;*/
+			return ret;
 		break;
 	case BMA_IOCTL_WRITE_CALI_VALUE:
 		pdata->gs_kvalue = kbuf;
@@ -344,8 +345,8 @@ static int bma_ioctl(struct inode *inode, struct file *file, unsigned int cmd,
 
 	switch (cmd) {
 	case BMA_IOCTL_READ:
-		/*if (copy_to_user(argp, &rwbuf, sizeof(rwbuf)))
-			return -EFAULT;*/
+		if (copy_to_user(argp, &rwbuf, sizeof(rwbuf)))
+			return -EFAULT;
 		break;
 	case BMA_IOCTL_READ_ACCELERATION:
 		if (copy_to_user(argp, &buf, sizeof(buf)))
@@ -555,7 +556,6 @@ int bma250_probe(struct i2c_client *client, const struct i2c_device_id *id)
 		err = -ENODEV;
 		goto exit_check_functionality_failed;
 	}
-
 	bma = kzalloc(sizeof(struct bma250_data), GFP_KERNEL);
 	if (!bma) {
 		err = -ENOMEM;
@@ -610,7 +610,7 @@ int bma250_probe(struct i2c_client *client, const struct i2c_device_id *id)
 		goto err_registerAttr;
 	}
 	D("%s:OK\n", __func__);
-	debug_flag = 0;
+	debug_flag = 1;
 
 	return 0;
 
diff --git a/drivers/i2c/chips/isl29029.c b/drivers/i2c/chips/isl29029.c
index 66d3e17..1c17018 100644
--- a/drivers/i2c/chips/isl29029.c
+++ b/drivers/i2c/chips/isl29029.c
@@ -1641,7 +1641,7 @@ static int isl29029_setup(struct isl29029_info *lpi)
 #ifdef POLLING_PROXIMITY
 		IRQF_TRIGGER_FALLING,
 #else
-		IRQF_TRIGGER_LOW,
+		IRQF_TRIGGER_FALLING,
 #endif
 		"isl29029",
 		lpi);
diff --git a/vendor/kernel-firmware/readme b/vendor/kernel-firmware/readme
deleted file mode 100644
index e69de29..0000000
